src/webui/api/searchcontroller.cpp

```diff
@@ -0,0 +1,384 @@
+        setResult(getResults(searchResults.mid(offset, limit), searchHandler->isActive(), size));
+    else
+        setResult(getResults(searchResults, searchHandler->isActive(), size));
+}
+
+void SearchController::deleteAction()
+{
+    checkParams({"id"});
+
+    const QString id = params()["id"].trimmed();
+    ISession *session = sessionManager()->session();
+
+    auto searchHandlers = session->getData(QLatin1String("searchHandlers")).value<SearchHandlerDict>();
+    if (!searchHandlers.contains(id))
+        throw APIError(APIErrorType::NotFound);
+
+    SearchHandlerPtr searchHandler = searchHandlers[id];
+    searchHandler->cancelSearch();
+    searchHandlers.remove(id);
+    session->setData(QLatin1String("searchHandlers"), QVariant::fromValue(searchHandlers));
+
+    auto activeSearches = session->getData(QLatin1String("activeSearches")).value<QSet<QString>>();
+    activeSearches.remove(id);
+    session->setData(QLatin1String("activeSearches"), QVariant::fromValue(activeSearches));
+}
+
+void SearchController::categoriesAction()
+{
+    QStringList categories;
+    const QString name = params()["pluginName"].trimmed();
+
+    categories << SearchPluginManager::categoryFullName("all");
+    for (const QString &category : SearchPluginManager::instance()->getPluginCategories(name))
+        categories << SearchPluginManager::categoryFullName(category);
+
+    const QJsonArray result = QJsonArray::fromStringList(categories);
+    setResult(result);
+}
+
+void SearchController::pluginsAction()
+{
+    const QStringList allPlugins = SearchPluginManager::instance()->allPlugins();
+    setResult(getPluginsInfo(allPlugins));
+}
+
+void SearchController::installPluginAction()
+{
+    checkParams({"sources"});
+
+    const QStringList sources = params()["sources"].split('|');
+    for (const QString &source : sources)
+        SearchPluginManager::instance()->installPlugin(source);
+}
+
+void SearchController::uninstallPluginAction()
+{
+    checkParams({"names"});
+
+    const QStringList names = params()["names"].split('|');
+    for (const QString &name : names)
+        SearchPluginManager::instance()->uninstallPlugin(name.trimmed());
+}
+
+void SearchController::enablePluginAction()
+{
+    checkParams({"names", "enable"});
+
+    const QStringList names = params()["names"].split('|');
+    const bool enable = Utils::String::parseBool(params()["enable"].trimmed(), false);
+
+    for (const QString &name : names)
+        SearchPluginManager::instance()->enablePlugin(name.trimmed(), enable);
+}
+
+void SearchController::updatePluginsAction()
+{
+    SearchPluginManager *pluginManager = SearchPluginManager::instance();
+
+    connect(pluginManager, &SearchPluginManager::checkForUpdatesFinished, this, &SearchController::checkForUpdatesFinished);
+    connect(pluginManager, &SearchPluginManager::checkForUpdatesFailed, this, &SearchController::checkForUpdatesFailed);
+    pluginManager->checkForUpdates();
+}
```

Comment:
Maybe return a confirmation?

Reply:
`pluginManager->checkForUpdates()` doesn't return anything, so this would be a larger refactor.